### DISCUSSION TITLE
QEMU setup instructions

### DISCUSSION THREAD

**Discussion by hassankhan** [ludwig-v/wireless-carplay-dongle-reverse-engineering#60] [Q&A] [answered]
Hi all,

This might be a silly question as I'm new to RE; but is there a way to run the official firmware inside QEMU?

/cc @ludwig-v @jsm174

**Comment by hassankhan** [comment, marked as answer]
Found some here: https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/blob/master/Custom_Firmware/Scripts/Dropbear/NOTES.md

**Reply by jsm174** [reply]
Yeh, I put that together, just so I would remember what I did. 

It won't let you run the firmware binaries, it's just for compiling.

**Reply by hassankhan** [reply]
Thanks @jsm174, I did figure that out last night 😀 

I used your script to get SSH on the device and managed to extract the kernel image. Now I need to figure out how to create a partitioned disk image, then see if I can get QEMU to use that.

**Reply by jsm174** [reply]
Very cool @hassankhan! Can't wait to hear how that goes!

**Reply by hassankhan** [reply]
Haha it sounds a lot cooler than it actually was, I just SSH'd in and ran `dd if=/dev/mtdblock1 of=/tmp/mtd1.bin`. Next step is to find out what I'm supposed to do with the `.bin` 😅
